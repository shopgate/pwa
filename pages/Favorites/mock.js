const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPrice: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPrice: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

