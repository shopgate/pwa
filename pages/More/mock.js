import configureStore from 'redux-mock-store';

const defaultState = {
  client: {
    isFetching: false,
    appVersion: '0-app',
    libVersion: '0-lib',
    codeBaseVersion: '0-code',
  },
  menu: {
    menusById: {
      quickLinks: {
        entries: [
          {
            url: '/page/test',
            label: 'CMS Test',
          },
        ],
      },
    },
  },
};

const userLoggedInState = {
  ...defaultState,
  user: {
    login: {
      isLoggedIn: true,
    },
    data: {
      firstName: 'John',
      lastName: 'Doe',
    },
  },
};

const userLoggedOutState = {
  ...defaultState,
  user: {
    login: {
      isLoggedIn: false,
    },
    data: {},
  },
};

/**
 * Mocked store.
 * @returns {Object}
 */
export const getLoggedInStore = () => configureStore()(userLoggedInState);

/**
 * Mocked store.
 * @returns {Object}
 */
export const getLoggedOutStore = () => configureStore()(userLoggedOutState);
