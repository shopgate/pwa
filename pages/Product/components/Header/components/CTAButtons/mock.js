export const mockedState = {
  product: {
    currentProduct: {
      productId: 'product_1',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
          stock: {
            orderable: true,
          },
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
        },
      },
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};

export const mockedVariantState = {
  product: {
    currentProduct: {
      productId: 'product_1',
      productVariantId: 'product_2',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
          stock: {},
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
          stock: {
            orderable: true,
          },
        },
      },
    },
    optionsByProductId: {
      product_1: null,
      product_2: null,
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};
