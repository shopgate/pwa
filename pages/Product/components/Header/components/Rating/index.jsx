import React from 'react';
import PropTypes from 'prop-types';
import appConfig from '@shopgate/pwa-common/helpers/config';
import RatingStars from '@shopgate/pwa-ui-shared/RatingStars';
import RatingCount from 'Components/Reviews/components/RatingCount';
import { container } from './style';
import connect from './connector';

/**
 * Scrolls page to reviews excerpt.
 */
const scrollToRating = () => {
  const reviewsExcerpt = document.getElementById('reviewsExcerpt');
  if (
    typeof reviewsExcerpt !== 'object'
    || !reviewsExcerpt
    || !reviewsExcerpt.offsetTop
    || !reviewsExcerpt.closest
    || !reviewsExcerpt.closest('article')
  ) {
    return;
  }

  reviewsExcerpt
    .closest('article')
    .scroll(0, reviewsExcerpt.offsetTop - 30);
};
/**
 * The Rating component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Rating = ({ rating }) => {
  if (!appConfig.hasReviews || !rating || !rating.count) {
    return null;
  }

  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div role="link" tabIndex={0} className={container} onClick={scrollToRating}>
      <RatingStars value={rating.average} display="big" />
      <RatingCount count={rating.count} prominent />
    </div>
  );
};

Rating.propTypes = {
  rating: PropTypes.shape(),
};

Rating.defaultProps = {
  rating: null,
};

export default connect(Rating);
