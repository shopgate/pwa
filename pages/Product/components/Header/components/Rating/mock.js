/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export { getElementById };
