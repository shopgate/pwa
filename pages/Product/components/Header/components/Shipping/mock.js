export const mockedStoreWithShippingPrice = {
  product: {
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: 1,
          currency: 'EUR',
        },
      },
    },
  },
};
export const mockedStoreWithFreeShipping = {
  product: {
    ...mockedStoreWithShippingPrice.product,
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: 0,
          currency: 'EUR',
        },
      },
    },
  },
};

export const mockedStoreWithUnknownShipping = {
  product: {
    ...mockedStoreWithShippingPrice.product,
    shippingByProductId: {
      fakeId: {
        shipping: {
          price: null,
          currency: 'EUR',
        },
      },
    },
  },
};
