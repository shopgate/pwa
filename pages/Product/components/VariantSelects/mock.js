/* eslint-disable extra-rules/no-single-line-objects */
/* eslint-disable object-property-newline */

const selection = [
  {
    id: '1',
    label: 'Coloras kdjakdsjaklsdj aksdj aksjdkajsdkajsdjaskdjaksdjkajsdaiweiriasjd',
    value: null,
    selected: false,
    disabled: false,
    values: [
      {
        id: '1', label: 'Pink', disabled: false, selected: false,
      },
      {
        id: '2', label: 'Black', disabled: true, selected: false,
      },
    ],
  },
  {
    id: '2',
    label: 'Size',
    value: '2',
    selected: true,
    disabled: false,
    values: [
      {
        id: '1', label: 'S', disabled: false, selected: false,
      },
      {
        id: '2', label: 'M', disabled: false, selected: true,
      },
      {
        id: '3', label: 'asjdaksdlkasdkalsklaeksoas iodialskdajsdjasdjkajdkajsd adshj', disabled: true, selected: false,
      },
      {
        id: '4', label: 'XS', disabled: false, selected: false,
      },
      {
        id: '5', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '6', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '7', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '8', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '9', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '10', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '11', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '12', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '13', label: 'XL', disabled: false, selected: false,
      },
    ],
  },
  {
    id: '3',
    label: 'FOOasodjaksd jkasjdkajsdk askdjaksdjaskd kasjkdjaksdjasjdk',
    value: 'Barasdhasjd kasjdkajskdjaksdjaksdjaksjdakjsdkajsd aksjdakjsdkajd kjdkajskdja',
    selected: false,
    disabled: true,
    values: [
      {
        id: '7', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '8', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '9', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '10', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '11', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '12', label: 'XL', disabled: false, selected: false,
      },
      {
        id: '13', label: 'XL', disabled: false, selected: false,
      },
    ],
  },
];

const selectionWithWarning = [
  {
    id: '1',
    label: 'Color',
    value: null,
    selected: false,
    disabled: false,
    values: [
      {
        id: '1', label: 'Red', disabled: false, selected: false, availability: {
          text: 'Only 1 left',
          state: 'warning',
        },
      },
      {
        id: '2', label: 'Blue', disabled: false, selected: false, availability: {
          text: 'Only 2 left',
          state: 'warning',
        },
      },
      {
        id: '3', label: 'Green', disabled: false, selected: false, availability: {
          text: 'Only 3 left',
          state: 'warning',
        },
      },
      {
        id: '4', label: 'Pink', disabled: false, selected: false, availability: {
          text: 'Available',
          state: 'ok',
        },
      },
    ],
  },
];

const selectionWithAlert = [
  {
    id: '1',
    label: 'Size',
    value: null,
    selected: false,
    disabled: false,
    values: [
      {
        id: '1', label: 'S', disabled: false, selected: false, availability: { text: 'Available', state: 'ok' },
      },
      {
        id: '2', label: 'M', disabled: false, selected: false, availability: { text: 'Out of stock', state: 'alert' },
      },
      {
        id: '3', label: 'L', disabled: false, selected: false, availability: { text: 'Out of stock', state: 'alert' },
      },
    ],
  },
];

export {
  selection,
  selectionWithWarning,
  selectionWithAlert,
};

/* eslint-enable extra-rules/no-single-line-objects */
/* eslint-enable object-property-newline */
