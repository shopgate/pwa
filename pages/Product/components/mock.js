import {
  basicProductState as bpState,
  productRouteMock,
  getShippingStateForId,
} from '@shopgate/pwa-common-commerce/product/mock';
import { getReviewsStateForId } from '@shopgate/pwa-common-commerce/reviews/mock';

const { productId } = 913;

const product = {
  ...bpState.product,
  ...getShippingStateForId(productId),
};
const basicProductState = {
  ...bpState,
  ...getReviewsStateForId(productId),
  product,
  router: {
    routing: false,
    stack: [{ ...productRouteMock }],
  },
  favorites: {
    products: {
      lastChange: 0,
      lastFetch: 0,
      expires: 99999999999999,
      ids: [],
      isFetching: false,
      ready: true,
    },
  },
};

export { basicProductState };
