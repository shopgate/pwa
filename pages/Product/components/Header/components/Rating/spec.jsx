import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import {
  mockedStateWithTwoReviews,
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';
import Rating from './index';
import { getElementById, setMocks } from './mock';

describe('Rating (product header)', () => {
  const mockedStore = configureStore();

  /**
   * Makes component.
   * @param {Object} state State
   * @returns {Object}
   */
  const getComponent = state => mount(
    <Provider store={mockedStore(state)}>
      <Rating />
    </Provider>,
    mockRenderOptions
  );

  describe('Rendering', () => {
    it('should render rating when data is available', () => {
      setMocks();
      const component = getComponent(mockedStateWithTwoReviews);
      expect(component).toMatchSnapshot();
    });

    it('should render nothing when data is not available', () => {
      setMocks();
      const component = getComponent(mockedStateWithoutReview);
      expect(component.html()).toBe(null);
    });

    it('should render nothing when reviews are deactivated', () => {
      setMocks(false);
      const component = getComponent(mockedStateWithoutReview);
      expect(component.html()).toBe(null);
    });
  });

  describe('Scroll on click', () => {
    const scrollSpy = jest.fn();

    it('should scroll to reviews when clicked', () => {
      jest.spyOn(document, 'getElementById').mockImplementation(getElementById(scrollSpy));
      const component = getComponent(mockedStateWithTwoReviews);
      component.simulate('click');
      expect(scrollSpy.mock.calls[0][0]).toBe(0);
      expect(scrollSpy.mock.calls[0][1]).toBe(70);
      expect(scrollSpy).toHaveBeenCalled();
      document.getElementById.mockReset();
      document.getElementById.mockRestore();
    });

    it('should do nothing when clicked but no reviews excerpt element', () => {
      jest.spyOn(document, 'getElementById').mockImplementation(() => null);
      const component = getComponent(mockedStateWithTwoReviews);
      component.simulate('click');
      expect(scrollSpy.mock.calls.length).toBe(1);
      document.getElementById.mockReset();
      document.getElementById.mockRestore();
    });
  });
});

