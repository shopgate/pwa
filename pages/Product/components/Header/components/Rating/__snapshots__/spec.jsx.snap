// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating (product header) Rendering should render rating when data is available 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Rating)>
    <Rating
      dispatch={[Function]}
      rating={
        Object {
          "average": 50,
          "count": 4,
        }
      }
    >
      <div
        className="css-1w47iqp"
        onClick={[Function]}
        role="link"
        tabIndex={0}
      >
        <RatingStars
          className=""
          display="big"
          isSelectable={false}
          onSelection={[Function]}
          value={50}
        >
          <div
            aria-label="reviews.rating_stars"
            className="css-13azwyo "
            data-test-id="ratedStars: 2.5"
          >
            <div
              className="css-4s85e9"
            >
              <div
                className="css-1xicacy css-q4owmg"
                key="1"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="2"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="3"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="4"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="5"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
            </div>
            <div
              className="css-16y9zin"
            >
              <div
                className="css-1xicacy css-q4owmg"
                key="6"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="7"
              >
                <Star
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className="css-1xicacy css-q4owmg"
                key="2"
              >
                <StarHalfIcon
                  className=""
                  color={null}
                  size="1.24em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<defs><path id=\\"a\\" d=\\"M0 0h24v24H0z\\"/></defs>
  <clipPath id=\\"b\\"><use xlink:href=\\"#a\\" overflow=\\"visible\\"/></clipPath>
  <path clip-path=\\"ul(#b)\\" d=\\"M12 2L9.2 8.6 2 9.2 7.5 14l-1.6 7 6.2-3.7m-.1-1.9V6.1m2.8 7.3\\"/>"
                    size="1.24em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<defs><path id=\\"a\\" d=\\"M0 0h24v24H0z\\"/></defs>
  <clipPath id=\\"b\\"><use xlink:href=\\"#a\\" overflow=\\"visible\\"/></clipPath>
  <path clip-path=\\"ul(#b)\\" d=\\"M12 2L9.2 8.6 2 9.2 7.5 14l-1.6 7 6.2-3.7m-.1-1.9V6.1m2.8 7.3\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "1.24em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </StarHalfIcon>
              </div>
            </div>
          </div>
        </RatingStars>
        <RatingCount
          count={4}
          prominent={true}
        >
          <Translate
            className="css-oixusm"
            params={
              Object {
                "count": 4,
              }
            }
            string="reviews.review_count"
          >
            <span
              className="css-oixusm"
            >
              reviews.review_count
            </span>
          </Translate>
        </RatingCount>
      </div>
    </Rating>
  </Connect(Rating)>
</Provider>
`;
